source/fem-dem/gls_vans.cc

```diff
@@ -429,12 +449,426 @@ GLSVANSSolver<dim>::assemble_L2_projection_void_fraction()
+
+              quadrature_point_location =
+                fe_values_void_fraction.get_quadrature_points();
+
+              // Active Neighbors include the current cell as well
+              auto active_neighbors =
+                LetheGridTools::find_cells_around_cell<dim>(vertices_to_cell,
+                                                            cell);
+
+              for (unsigned int q = 0; q < n_q_points; ++q)
+                {
+                  particles_volume_in_sphere = 0;
+                  quadrature_void_fraction   = 0;
+                  std::vector<double> distance_to_face_vector;
+
+                  for (unsigned int m = 0; m < active_neighbors.size(); m++)
+                    {
+                      // Loop over particles in neighbor cell
+                      // Begin and end iterator for particles in neighbor cell
+
+                      const auto pic =
+                        particle_handler.particles_in_cell(active_neighbors[m]);
+                      for (auto &particle : pic)
+                        {
+                          double distance          = 0;
+                          auto particle_properties = particle.get_properties();
+                          double r_particle =
+                            particle_properties[DEM::PropertiesIndex::dp] / 2;
+                          double single_particle_volume =
+                            M_PI * pow((r_particle * 2), dim) / (2 * dim);
+
+                          // Distance between particle and quadrature point
+                          // centers
+                          distance = particle.get_location().distance(
+                            quadrature_point_location[q]);
+
+                          // Particle completely in reference sphere
+                          if (distance <= (R_sphere - r_particle))
+                            particles_volume_in_sphere +=
+                              (M_PI *
+                               pow(
+                                 particle_properties[DEM::PropertiesIndex::dp],
+                                 dim) /
+                               (2 * dim)) *
+                              single_particle_volume /
+                              particle_properties
+                                [DEM::PropertiesIndex::contribution];
+
+                          // Particle completely outside reference sphere
+                          else if (distance >= (R_sphere + r_particle))
+                            particles_volume_in_sphere += 0;
+
+                          // Particle partially in reference sphere
+                          else if ((distance > (R_sphere - r_particle)) &&
+                                   (distance < (R_sphere + r_particle)))
+                            {
+                              if (dim == 2)
+                                {
+                                  particles_volume_in_sphere +=
+                                    (pow(r_particle, 2) *
```

Comment:
I suggest adding these calculations for 2d and 3d into two separate functions and calling the functions (here and above).